### DISCUSSION TITLE
Why TypeScript shows that data from useAsyncData can be null?

### DISCUSSION THREAD

**Discussion by rudimadima** [nuxt/nuxt#16389] [Questions]
Hi. Simple example:
```vue
<template>
  <div>
    {{ data.someText }}
  </div>
</template>

<script setup lang="ts">
const { data } = await useAsyncData(
  `test`,
  () => {
    return new Promise<{ someText: string }>((resolve) => {
      resolve({
        someText: 'hi',
      });
    });
  },
  {
    initialCache: false,
  }
);
</script>

```
This has never happened before. 
But now the TypeScript says that `data` from can be null. I updated my Nuxt to rc.11 and now this problem.
Why? How it works?

**Comment by noook** [comment]
I'm upping this discussion: null means the data is possibly not available yet I guess, however with SSR and the await in the setup, it should already be available, and thus not being null.

Is there an explanation to this ?

Edit: I guess this is the answer: https://github.com/nuxt/framework/issues/8747
`data` can be null in case of an error
